runtime/lib/ttmetal/command_queue.cpp

```diff
@@ -67,6 +74,47 @@ CQExecutor::CQExecutor(::tt::tt_metal::Device *device, std::size_t cq_id,
   }
 
   cq = &device->command_queue(cq_id);
+
+  createReservedL1Buffer();
+}
+
+void CQExecutor::createReservedL1Buffer() {
```

Comment:
I'm actually not sure any runtime changes are needed.  Shouldn't we get the address/cb info from the compiler just like all the others?

Reply:
As I said about `is_internal` attribute, there is no tensor ref coming from the compiler that represents this buffer. 

I decided not to explicitly create tensor in the graph that represents scaler CB when lowering to a generic op. This is because it gets tricky creating layouts and empty ops before TTIRLayoutPass that is normally done after generic op lowering phase. Plus, logically it seems better not to pollute TTIR graph with this kind of kernel intrinsics.

Therefore, I put this code in the runtime to allocate small buffer at the top of L1. 

Btw I noticed in metal, there is notion of `BufferType::L1_SMALL` suitable for this kind of stuff, right? Is it only used for metal internal allocations?

Reply:
> Can we just encode this address directly on CBRef.address in the flatbuffer? 

At first I thought this was not possible due to tt-metal API that doesn't allow you to create such `CircularBufferConfig` with local address and without shadow buffer. 
But on the other look, I think it can be done but in 3 steps.
1. Create CB without shadow buffer.
2. Get CB ptr with CB handle from 1. 
3. Update it using `set_locally_allocated_address(uint32_t address)`

I will try these steps for internal CBs. If this works I will remove creating this shadow buffer in the scratch space.